common/src/test/java/com/microsoft/identity/common/MsalCppOAuth2TokenCacheTest.java

```diff
@@ -182,6 +183,97 @@ public void removeAccountTest() throws ClientException {
         Assert.assertNull(restoredAccount);
     }
 
+    @Test
+    public void removeNoMatchingAccount() throws ClientException {
+        // Get the generated account
+        final AccountRecord generatedAccount = mTestBundle.mGeneratedAccount;
+
+        // Save it to the cache
+        mCppCache.saveAccountRecord(generatedAccount);
+        mCppCache.saveCredentials(null, mTestBundle.mGeneratedRefreshToken);
+
+        // Call remove with a different realm
+        final AccountDeletionRecord deletionRecord = mCppCache.removeAccount(
+                generatedAccount.getHomeAccountId(),
+                generatedAccount.getEnvironment(),
+                REALM2
+        );
+
+        // Check the receipt, should remove nothing
+        Assert.assertEquals(0, deletionRecord.size());
```

Comment:
may want to also add a check to actually query the cache and check that the item is still there...this check only asserts that `removeAccount` API returned correctly...it doesn't actually assert that the account wasn't truly removed

Reply:
Sure, will update.